linked_list/linked_list.py

```diff
@@ -9,71 +12,150 @@ def __init__(self, value, next_node = None):
+                return length
+        else:
+            return length
+
 
     # method that returns the value at a given index in the linked list
     # index count starts at 0
     # returns None if there are fewer nodes in the linked list than the index value
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def get_at_index(self, index):
-        pass
+        # use counter, when counter matches index, return
+        current_node = self.head
+        counter = 0
+        if not current_node:
+            return None
+        elif counter == index:
+            return current_node.value
+        while counter != index:
+            if current_node.next:
+                current_node = current_node.next
+            else:
+                return None
+            counter += 1
+        else:
+            return current_node.value
 
     # method that returns the value of the last node in the linked list
     # returns None if the linked list is empty
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(1) ??
     def get_last(self):
-        pass
+        current_node = self.head
+        if not current_node:
+            return None
+        while current_node.next:
+            current_node = current_node.next
+        else:
+            return current_node.value
 
     # method that inserts a given value as a new last node in the linked list
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def add_last(self, value):
-        pass
+        current_node = self.head
+        new_node = Node(value)
+        if not current_node:
+            self.head = new_node
+            return
+            # self.add_first(value)
+            # return
+        while current_node.next:
+            current_node = current_node.next
+        else:
+            current_node.next = new_node
 
     # method to return the max value in the linked list
     # returns the data value and not the node
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def find_max(self):
-        pass
+        current_node = self.head
+        if not current_node:
+            return None
+        current_max = -900000000
+        while current_node != None:
+            if current_max < current_node.value:
+                current_max = current_node.value
+            current_node = current_node.next
+        return current_max
+
 
     # method to delete the first node found with specified value
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def delete(self, value):
-        pass
+        current_node = self.head
+        if not current_node:
+            return None
+        while current_node:
+            if current_node.value == value:
+                self.head = current_node.next
+                break

```

Comment:
🤓I would suggest using return over break statements
```suggestion
            if current_node.value == value:
                self.head = current_node.next
                return

```